### PR TITLE
feat: upgrade eslint

### PR DESCRIPTION
**Pull request by moroine**
**Changes:**
- Upgrade eslint 8 & disable not maintained plugins.
- Re-order rule list as per plugins documentations
- Add new rules from plugin doc as `off`

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -6,10 +6,10 @@ module.exports = {
     './rules/style',
     './rules/variables',
     './rules/es6',
-    './rules/imports',
+    './rules/plugins/imports',
     './rules/strict',
-    './rules/flow.js',
-    './rules/eslint-comments',
+    './rules/plugins/flow',
+    // './rules/plugins/eslint-comments',
```
Comment (limtingzhi) [inline base.js:12]: Won't this be useful to remove unused comments?

Comment (moroine) [reply base.js:12]: The plugin is not supported anymore. But I think we can use https://eslint.org/docs/latest/use/configure/rules#report-unused-eslint-disable-comments instead

**Review by limtingzhi — APPROVED** [review]

**Pull request merged**